mongoOperator/helpers/listeners/mongo/HeartbeatListener.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2018 Ultimaker
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+import logging
+from typing import Callable, Dict
+
+from pymongo.monitoring import ServerHeartbeatListener, ServerHeartbeatStartedEvent, ServerHeartbeatSucceededEvent,\
+    ServerHeartbeatFailedEvent
+
+from mongoOperator.models.V1MongoClusterConfiguration import V1MongoClusterConfiguration
+
+
+class HeartbeatListener(ServerHeartbeatListener):
+    """ A listener for Mongo server heartbeats. """
+
+    INVALID_REPLICA_SET_CONFIG = "Does not have a valid replica set config"
+
+    def __init__(self, cluster_object: V1MongoClusterConfiguration,
+                 all_hosts_ready_callback: Callable[[V1MongoClusterConfiguration], None]) -> None:
+        super().__init__()
+        self._cluster_object: V1MongoClusterConfiguration = cluster_object
+        self._expected_host_count: int = cluster_object.spec.mongodb.replicas
+        self._hosts: Dict[str, int] = {}
```

Comment:
What is this dict supposed to contain?

Reply:
A key (hostname of a cluster node) and a value (1) indicating the host is ready.